src/main/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupport.scala

```diff
@@ -176,6 +225,60 @@ trait DataUseRestrictionSupport extends LazyLogging {
       allDurFieldNames.map(AttributeName.withLibraryNS)) ++ preferred
   }
 
+  def generateStructuredUseRestrictionAttribute(request: StructuredDataRequest, ontologyDAO: OntologyDAO): Map[String, JsValue] = {
+    // get DS diseases (map over array of ints)
+    val diseaseCodesArray = getDiseaseNames(request.diseaseUseOnly, ontologyDAO).map("DS:" + _)
+
+    // create map of correct restrictions
+    val consentMap = Map(
+      AttributeName.withDefaultNS(ConsentCodes.GRU) -> AttributeBoolean(request.generalResearchUse),
+      AttributeName.withDefaultNS(ConsentCodes.HMB) -> AttributeBoolean(request.healthMedicalBiomedicalUseOnly),
+      AttributeName.withDefaultNS(ConsentCodes.NCU) -> AttributeBoolean(request.commercialUseProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NPU) -> AttributeBoolean(request.forProfitUseProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NMDS) -> AttributeBoolean(request.methodsResearchProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NAGR) -> AttributeBoolean(request.aggregateLevelDataProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NCTRL) -> AttributeBoolean(request.controlsUseProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.RSPD) -> AttributeBoolean(request.pediatricResearchOnly),
+      AttributeName.withDefaultNS(ConsentCodes.IRB) -> AttributeBoolean(request.IRB)) ++ getGenderCodeMap(request.genderUseOnly)
+
+    // convert to array of consent codes
+    val consentCodes = consentMap.filter(_._2.value).map(_._1.name).toArray ++ diseaseCodesArray
+
+    //formatWithPrefix(request.prefix, StructuredDataResponse(consentCodes, 1.0, consentMap ++ Map(ConsentCodes.DS -> AttributeValueList(request.diseaseUseOnly.map(AttributeNumber(_))))))
+    StructuredDataResponse(consentCodes, 1.0, request.prefix, consentMap ++ Map(AttributeName.withDefaultNS(ConsentCodes.DS) -> AttributeValueList(request.diseaseUseOnly.map(AttributeNumber(_))))).formatWithPrefix
+  }
+
+  private def getDiseaseNames(diseaseCodes: Array[Int], ontologyDAO: OntologyDAO): Array[String] = {
+    getDiseaseNames(diseaseCodes.map("http://purl.obolibrary.org/obo/DOID_" + _.toString), ontologyDAO)
+  }
+
+  private def getDiseaseNames(diseaseCodes: Array[String], ontologyDAO: OntologyDAO): Array[String] = {
+   diseaseCodes.map { nodeid =>
+      ontologyDAO.search(nodeid) match {
+        case termResource :: Nil => termResource.label
+        case _ =>  "" // return an error if the int does not correlate to a disease?
+      }
+    }
+  }
+
+
+  private def getGenderCodeMap(rsg: String): Map[AttributeName, AttributeBoolean] = {
+    rsg.toLowerCase match {
+      case "female" =>
+        Map(AttributeName.withDefaultNS(ConsentCodes.RSG) -> AttributeBoolean(true),
+          AttributeName.withDefaultNS(ConsentCodes.RSFM) -> AttributeBoolean(true),
+          AttributeName.withDefaultNS(ConsentCodes.RSM) -> AttributeBoolean(false))
```

Comment:
Setting a false value goes back to an earlier comment - what is the effect of a consent code having a false value vs no value at all? Do we use false values in any way? If so, then this makes sense. If not, then it's just extraneous data that we don't need to store.